### PR TITLE
Adding iterative least squares for outlier removal.

### PR DESCRIPTION
**Pull request by bryanvriel**
Added new function for spatial gradient computation (`method='robust_l2'`) that iteratively removes outliers for least squares solution.

@maxfilter I went ahead and added this functionality since I needed it for one of my projects. Your original reweighted routine is accessible via `method='robust_lp'`, with common entrypoints at `ice.gradient` and `ice.robust_gradient`.

### CODE REVIEW

**Review by maxfilter — COMMENTED** [review]
Looks great! I haven't tested it out on a synthetic dataset yet, but might try doing that later tonight.

File: iceutils/stress.py [hunk 1]
```diff
@@ -498,6 +511,95 @@ def _run_IRLS(Z, G, half_sizes, ftol, maxiter, coord):
...
+    Z : array_like
+        2-dimensional array of data with padding
+    G : array_like
+        2-dimensional design matrix of shape (window_size^2, ncoef)
+    half_sizes: tuple of ints
+        half the window length in both directions (half_size_y, half_size_x).
+    std_thresh: float
+        Number of standard deviations above which defines outliers.
+    maxiter : int
+        maximum number of iterations of IRLS when approximating the L1 norm.
```
[the remark this answers is not in the document]
Comment (bryanvriel) [reply iceutils/stress.py]: Nice catch.

File: iceutils/stress.py [hunk 2]
```diff
@@ -498,6 +511,95 @@ def _run_IRLS(Z, G, half_sizes, ftol, maxiter, coord):
...
+        dz/dy   @ (0, 0) = a1
+        dz/dx   @ (0, 0) = a2
+        etc...
+    """
+    # Cache number of parameters
+    N_par = G.shape[1]
+
+    # Don't fit if the center coordinate is nan
+    if np.isnan(Z[coord]):
+        return [np.nan]*len(G[0])
```
[the remark this answers is not in the document]
Comment (bryanvriel) [reply iceutils/stress.py]: Yes, another nice catch!

**Comment by maxfilter** [comment]
robust_l2_test

Wasn't sure if you wanted me to merge or not, but I had a chance to test it with that script you sent me and it looks great!

**Comment by bryanvriel** [comment]
Cool, glad to see it works!

**Pull request merged**